v1.15.1 still panics in GetMaintenanceResultFromJob

**What steps did you take and what happened:**

I upgraded to v1.15.1 and use the kopia uploader. The velero pod restarts due to panic

```
time="2024-12-29T16:36:43Z" level=info msg="Running maintenance on backup repository" backupRepo=velero/p-project-default-kopia-6bmb6 logSource="pkg/controller/backup_repository_controller.go:309"
time="2024-12-29T16:36:43Z" level=info msg="Start to maintenance repo" BSL name=default logSource="pkg/repository/manager/manager.go:216" repo UID=1183e359-8bb9-4240-af20-ae3a64f09756 repo name=p-project-default-kopia-6bmb6 repo type=kopia
time="2024-12-29T16:36:48Z" level=info msg="Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference" BackupRepository="{\"name\":\"p-project-default-kopia-6bmb6\",\"namespace\":\"velero\"}" controller=backuprepository controllerGroup=velero.io controllerKind=BackupRepository logSource="/go/pkg/mod/github.com/bombsimon/logrusr/v3@v3.0.0/logrusr.go:108" name=p-project-default-kopia-6bmb6 namespace=velero reconcileID="\"48e60053-612f-4651-ba3c-bdf4b53a6b12\""
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1843d63]

goroutine 1439 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x284a160?, 0x478c5a0?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/vmware-tanzu/velero/pkg/repository.GetMaintenanceResultFromJob({0x3198a80, 0xc00040f200}, 0xc0043e4a08)
	/go/src/github.com/vmware-tanzu/velero/pkg/repository/maintenance.go:120 +0x1a3
github.com/vmware-tanzu/velero/pkg/repository/manager.(*manager).PruneRepo(0xc0004b8000, 0xc00413e000)
	/go/src/github.com/vmware-tanzu/velero/pkg/repository/manager/manager.go:249 +0xb4b
github.com/vmware-tanzu/velero/pkg/controller.(*BackupRepoReconciler).runMaintenanceIfDue(0xc000533110, {0x3182ca0, 0xc00412b6e0}, 0xc00413e000, {0x31a4ae0, 0xc0041093b0})
	/go/src/github.com/vmware-tanzu/velero/pkg/controller/backup_repository_controller.go:315 +0x142
github.com/vmware-tanzu/velero/pkg/controller.(*BackupRepoReconciler).Reconcile(0xc000533110, {0x3182ca0, 0xc00412b6e0}, {{{0xc001cf3e76?, 0x0?}, {0xc0028741b0?, 0xc00412b6e0?}}})
	/go/src/github.com/vmware-tanzu/velero/pkg/controller/backup_repository_controller.go:208 +0x606
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x318aee0?, {0x3182ca0?, 0xc00412b6e0?}, {{{0xc001cf3e76?, 0xb?}, {0xc0028741b0?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00013f0e0, {0x3182cd8, 0xc000957220}, {0x29ad600, 0xc000ba0400})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:316 +0x3bc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00013f0e0, {0x3182cd8, 0xc000957220})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:266 +0x1be
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 1066
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:223 +0x50c
````

**Anything else you would like to add:**
<!--Miscellaneous information that will assist in solving the issue.-->




**Environment:**

- Velero version (use `velero version`):

```
velero version
Client:
	Version: v1.15.0
	Git commit: none
Server:
	Version: v1.15.1
```

- Velero features (use `velero client config get features`):

```
features: <NOT SET>
```

- Kubernetes version (use `kubectl version`):

`v1.22.8`

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

Seems  these two are not the same thing.Here is my error message


panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x227dce3]

goroutine 1410 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x281c120?, 0x4727010?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/vmware-tanzu/velero/pkg/repository.getMaintenanceResultFromJob({0x315aea0, 0xc000629950}, 0xc001b9b908)
	/go/src/github.com/vmware-tanzu/velero/pkg/repository/maintenance.go:238 +0x1a3
github.com/vmware-tanzu/velero/pkg/repository.(*manager).PruneRepo(0xc00082f550, 0xc004413500)
	/go/src/github.com/vmware-tanzu/velero/pkg/repository/manager.go:234 +0xacc
github.com/vmware-tanzu/velero/pkg/controller.(*BackupRepoReconciler).runMaintenanceIfDue(0xc000e1b2c0, {0x31450e0, 0xc00425b740}, 0xc004413500, {0x3166de0, 0xc000635a40})
	/go/src/github.com/vmware-tanzu/velero/pkg/controller/backup_repository_controller.go:299 +0x142
github.com/vmware-tanzu/velero/pkg/controller.(*BackupRepoReconciler).Reconcile(0xc000e1b2c0, {0x31450e0, 0xc00425b740}, {{{0xc001a8cde6?, 0x0?}, {0xc001a90540?, 0xc00425b740?}}})
	/go/src/github.com/vmware-tanzu/velero/pkg/controller/backup_repository_controller.go:201 +0x606
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x314d298?, {0x31450e0?, 0xc00425b740?}, {{{0xc001a8cde6?, 0xb?}, {0xc001a90540?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000a3c960, {0x3145118, 0xc0008754a0}, {0x297d680, 0xc0042af5a0})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:316 +0x3bc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000a3c960, {0x3145118, 0xc0008754a0})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:266 +0x1be
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 1146
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:223 +0x50c

--- a/changelogs/unreleased/8568-mcluseau
+++ b/changelogs/unreleased/8568-mcluseau
@@ -1,0 +1,1 @@
+fix(pkg/repository/maintenance): don't panic when there's no container statuses

--- a/pkg/repository/maintenance.go
+++ b/pkg/repository/maintenance.go
@@ -117,7 +117,20 @@ func GetMaintenanceResultFromJob(cli client.Client, job *batchv1.Job) (string, e
 	}
 
 	// we only have one maintenance pod for the job
-	return podList.Items[0].Status.ContainerStatuses[0].State.Terminated.Message, nil
+	pod := podList.Items[0]
+
+	statuses := pod.Status.ContainerStatuses
+	if len(statuses) == 0 {
+		return "", fmt.Errorf("no container statuses found for job %s", job.Name)
+	}
+
+	// we only have one maintenance container
+	terminated := statuses[0].State.Terminated
+	if terminated == nil {
+		return "", fmt.Errorf("container for job %s is not terminated", job.Name)
+	}
+
+	return terminated.Message, nil
 }
 
 func GetLatestMaintenanceJob(cli client.Client, ns string) (*batchv1.Job, error) {

--- a/pkg/repository/maintenance_test.go
+++ b/pkg/repository/maintenance_test.go
@@ -188,33 +188,54 @@ func TestGetMaintenanceResultFromJob(t *testing.T) {
 		},
 	}
 
-	// Set up test pod
+	// Set up test pod with no status
 	pod := &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "test-pod",
 			Namespace: "default",
 			Labels:    map[string]string{"job-name": job.Name},
 		},
-		Status: v1.PodStatus{
-			ContainerStatuses: []v1.ContainerStatus{
-				{
-					State: v1.ContainerState{
-						Terminated: &v1.ContainerStateTerminated{
-							Message: "test message",
-						},
+	}
+
+	// Create a fake Kubernetes client
+	cli := fake.NewClientBuilder().WithObjects(job, pod).Build()
+
+	// test an error should be returned
+	result, err := GetMaintenanceResultFromJob(cli, job)
+	assert.Error(t, err)
+	assert.Equal(t, "", result)
+
+	// Set a non-terminated container status to the pod
+	pod.Status = v1.PodStatus{
+		ContainerStatuses: []v1.ContainerStatus{
+			{
+				State: v1.ContainerState{},
+			},
+		},
+	}
+
+	// Test an error should be returned
+	cli = fake.NewClientBuilder().WithObjects(job, pod).Build()
+	result, err = GetMaintenanceResultFromJob(cli, job)
+	assert.Error(t, err)
+	assert.Equal(t, "", result)
+
+	// Set a terminated container status to the pod
+	pod.Status = v1.PodStatus{
+		ContainerStatuses: []v1.ContainerStatus{
+			{
+				State: v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{
+						Message: "test message",
 					},
 				},
 			},
 		},
 	}
 
-	// Create a fake Kubernetes client
-	cli := fake.NewClientBuilder().WithObjects(job, pod).Build()
-
-	// Call the function
-	result, err := GetMaintenanceResultFromJob(cli, job)
-
-	// Check if the result and error match the expectation
+	// This call should return the termination message with no error
+	cli = fake.NewClientBuilder().WithObjects(job, pod).Build()
+	result, err = GetMaintenanceResultFromJob(cli, job)
 	assert.NoError(t, err)
 	assert.Equal(t, "test message", result)
 }

